geode-redis/src/main/java/org/apache/geode/redis/internal/executor/hash/GeodeRedisHashSynchronized.java

```diff
@@ -89,6 +89,11 @@ public int hdel(List<ByteArrayWrapper> subList) {
       numDeleted.set(oldHash.size() - newHash.size());
       return newHash;
     });
+
+    if (hgetall().isEmpty()) {
+      RedisDataType type = context.getKeyRegistrar().getType(key);
```

Comment:
It seems wrong here to refetch the RedisDataType and then blow it away without checking if the type is what we expect. For example here should we check to see if it is == REDIS_HASH before calling removeKey?

Reply:
Turns out the type was not getting checked for HDel, so we added a test and updated the HDelExecutor.  Thanks for pointing this out!